packages/react-dom/src/client/ReactDOMComponentTree.js

```diff
@@ -47,11 +47,35 @@ export function detachDeletedInstance(node: Instance): void {
   // these fields are relevant.
   delete (node: any)[internalInstanceKey];
   delete (node: any)[internalPropsKey];
+  delete (node: any)[internalContainerInstanceKey];
```

Comment:
this was something we also cleared out in the previous v17 fix, however its not needed imo. This is only set for the container root node (which I _think_ means the root of react's tree). since v17, you can have _multiple_ roots so react can run multiple versions, but I think we only have one root.